### ISSUE TITLE
[call-overload] `Series.name` is incompatible with `DataFrame.__getitem__`

### ISSUE THREAD

**Issue by randolf-scholz** [pandas-dev/pandas-stubs#570]
**Describe the bug**

It should be type-safe to put `Series.name` into `DataFrame.__getitem__`, but `Series.name` is defined as `None | Hashable` and `DataFrame.__getitem__` does not accept `Hashable`.

https://github.com/pandas-dev/pandas-stubs/blob/d2e3f74e20ddbaf8375e29f84cbaa8a139cf8353/pandas-stubs/core/frame.pyi#L510-L523

https://github.com/pandas-dev/pandas-stubs/blob/d2e3f74e20ddbaf8375e29f84cbaa8a139cf8353/pandas-stubs/core/series.pyi#L326-L327

**To Reproduce**

```python
from pandas import DataFrame, Series

foo = Series([1, 2, 3], name="foo")
bar = DataFrame(columns=["foo", "bar", "baz"])
bar[foo.name]  # [call-overload]
# No overload variant of "__getitem__" of "DataFrame" matches argument type "Hashable" 
```

**Please complete the following information:**
- pandas                        1.5.3
- pandas-stubs                  1.5.3.230304

**Comment by Dr-Irv** [comment]
We can change 
```python
 def __getitem__(self, idx: Scalar | tuple[Hashable, ...]) -> Series: ... 
```
to
```python
 def __getitem__(self, idx: Hashable| tuple[Hashable, ...]) -> Series: ... 
```
but you'd still have to write `bar[cast(Hashable, foo.name)]` because `foo.name` could be `None`

**Comment by randolf-scholz** [comment]
@Dr-Irv `None` is an instance of `Hashable` so it should work without casting.

**Comment by Dr-Irv** [comment]
> @Dr-Irv `None` is an instance of `Hashable` so it should work without casting.

Fair point - so we should change the return type of `Series.name` as well.

**Comment by Dr-Irv** [comment]
This seems to be fixed as of version 2.0.1.230501.
